app/models/asserted_distribution.rb

```diff
@@ -175,6 +175,73 @@ def self.batch_update(params)
     query_batch_update(request)
   end
 
+  def self.batch_template_create(params)
```

Comment:
I feel ikely ultimately we should move these batch related code to their own <something>Batch modules., then use constants for the cutoff values etc. For later...